src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -670,6 +677,30 @@ public static function getHttpMethodParameterOverride(): bool
         return self::$httpMethodParameterOverride;
     }
 
+    /**
+     * Sets the list of HTTP methods that can be overridden.
+     *
+     * Set to null to allow all methods to be overridden (default). Set to an
+     * empty array to disallow overrides entirely. Otherwise, provide the list
+     * of uppercased method names that are allowed.
+     *
+     * @param string[]|null $methods
```

Comment:
should we use `uppercase-string[]` here, so that both Psalm and phpstan report errors during static analysis if you pass them in lowercase ?